osquery/tables/networking/windows/interfaces.cpp

```diff
@@ -276,8 +277,13 @@ QueryData genInterfaceAddresses(QueryContext& context) {
 
   const IP_ADAPTER_ADDRESSES* currAdapter = adapters.get();
   while (currAdapter != nullptr) {
-    std::wstring wsAdapterName = std::wstring(currAdapter->FriendlyName);
-    auto adapterName = std::string(wsAdapterName.begin(), wsAdapterName.end());
+    const auto wAdapterName = currAdapter->FriendlyName;
+    const auto wAdapterLen = wcslen(wAdapterName);
+    std::string adapterName;
+    adapterName.reserve(wAdapterLen);
+    for (size_t idx = 0; idx < wAdapterLen; ++idx) {
+      adapterName.push_back(static_cast<char>(wAdapterName[idx]));
```

Comment:
Is it a good idea to convert `wchar_t` via static cast? Look dangerous to be honest. Could we do some proper conversion to `utf-8` instead?
```c++
    std::wstring_convert<std::codecvt_utf8<wchar_t>> conv;
    std::string u8str = conv.to_bytes(wstr);
```

Reply:
I preserved the exact behavior that was here before :). That operation being potentially dangerous is exactly why the warning is getting emitted. As far as I am aware wstring_convert and codecvt_utf8 doesn't actually produce a correct transform either, which is why that got deprecated in C++17. I know ways to solve that problem but unfortunately they're all platform specific.

(I don't know exactly what OSQuery does but it seems like you'd have to have a normal way to do this conversion somewhere?)

Reply:
We've tried to unify our narrow-to-wide [conversions here](https://github.com/facebook/osquery/blob/master/osquery/core/windows/wmi.h#L40-L52), but it seriously wouldn't surprise me to hear we've been doing this wrong, so any advice you could offer would be amazing! We have had to avoid the wide string APIs, as our rocksdb backing store is setup to do UTF-8 at the moment, as such we've always just had folks convert to narrow string equivalents before shipping data off.

Reply:
@muffins It looks like this problem is more pervasive than that I point out here; and those conversion functions are going to need to be called in a more widespread fashion. Is there some higher level header that should have that declaration or should I add #include wmi to things?

Reply:
Actually nvm it looks like wmi is already included in those places.